multiqc/core/ai.py

```diff
@@ -0,0 +1,764 @@
+        if section.description:
+            sec_context += f"Section description: {_strip_html(section.description)}\n"
+        if section.comment:
+            sec_context += f"Section comment: {_strip_html(section.comment)}\n"
+        if section.helptext:
+            sec_context += f"Section help text: {_strip_html(section.helptext)}\n"
+
+        if section.content_before_plot:
+            sec_context += section.content_before_plot + "\n\n"
+        if section.content:
+            sec_context += section.content + "\n\n"
+
+        if section.plot_anchor and section.plot_anchor in report.plot_by_id:
+            plot = report.plot_by_id[section.plot_anchor]
+            if plot_content := plot.format_for_ai_prompt(keep_hidden=True):
+                if plot.pconfig.title:
+                    sec_context += f"Title: {plot.pconfig.title}\n"
+                sec_context += "\n" + plot_content
+
+        # Check if adding this section would exceed the limit
+        # Using rough estimate of 4 chars per token
+        sec_n_tokens = client.n_tokens(sec_context)
+        if current_n_tokens + sec_n_tokens > max_tokens:
+            logger.warning(
+                f"Truncating prompt to only the general stats to fit within {client.title}'s context window ({client.max_tokens()} tokens). "
+                f"Tokens estimate: {current_n_tokens}, with sections: at least {current_n_tokens + sec_n_tokens}"
+            )
+            return user_prompt, False
+
+    user_prompt += sec_context
+    user_prompt = re.sub(r"\n\n\n", "\n\n", user_prompt)  # strip triple newlines
+    return user_prompt, False
+
+
+def _save_prompt_to_file(prompt: str):
+    # Save content to file for debugging
+    path = report.data_tmp_dir() / "multiqc_ai_prompt.txt"
+    system_prompt = PROMPT_FULL if config.ai_summary_full else PROMPT_SHORT
+    path.write_text(f"{system_prompt}\n\n----------------------\n\n{prompt}")
+    logger.debug(f"Saved AI prompt to {path}")
+
+
+def add_ai_summary_to_report():
+    metadata: AiReportMetadata = ai_section_metadata()
+    # Set data for JS runtime
+    report.ai_report_metadata_base64 = base64.b64encode(metadata.model_dump_json().encode()).decode()
+
+    if not config.ai_summary:
+        # Not generating report in Python, leaving for JS runtime
+        return
+
+    if not (client := get_llm_client()):
+        return
+
+    report.ai_provider_id = client.name
+    report.ai_provider_title = client.title
+    report.ai_model = client.model
+
+    prompt, exceeded_context_window = build_prompt(client, metadata)
+
+    _save_prompt_to_file(prompt)
+
+    if exceeded_context_window:
+        return
+
+    response: Optional[InterpretationResponse]
+    if config.ai_summary_full:
+        response = client.interpret_report_full(prompt)
+    else:
+        response = client.interpret_report_short(prompt)
+    if not response:
+        return None
+
+    logger.debug(f"Interpretation: {response.interpretation}")
```

Comment:
This could be quite long and it's also in the report, so not sure it's worth a debug log?
```suggestion
```